dart/lib/src/hub.dart

```diff
@@ -317,7 +318,7 @@ class Hub {
   }
 
   /// Configures the scope through the callback.
-  void configureScope(ScopeCallback callback) {
+  FutureOr<void> configureScope(ScopeCallback callback) {
```

Comment:
`Sentry.configureScope` also has to change, not only the `Hub` class, likely the NoOp classes, etc.

Reply:
Maybe `Scope#clone` also has to change since it calls the setters that are `Future`.
I'd double-check every interaction with the methods that are now `Future` or `FutureOr`

Reply:
I also found this https://github.com/getsentry/sentry-dart/issues/926 but its a different issue and can be addressed in a different PR